File: CS/SimpleCustomFields/MyAppointmentForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
#region #usingsform
using DevExpress.XtraScheduler.UI;
using DevExpress.XtraScheduler;
#endregion #usingsform

namespace SimpleCustomFields {
    public partial class MyAppointmentForm : DevExpress.XtraScheduler.UI.AppointmentForm {
        public MyAppointmentForm(SchedulerControl control, Appointment apt) : base(control, apt) {
            // Required to initialize custom field editors.
            InitializeComponent();
        }

        public MyAppointmentForm(SchedulerControl control, Appointment apt, bool openRecurrenceForm) : base(control, apt, openRecurrenceForm)
        {
            // Required to initialize custom field editors.
            InitializeComponent();
        }

        public override void LoadFormData(Appointment apt)
        {
            base.LoadFormData(apt);
            UpdateCustomFieldEditors();
        }
        #region #Form-Controller
        protected override AppointmentFormController CreateController(SchedulerControl control, Appointment apt) {
            return new MyAppointmentFormController(control, apt);
        }

        protected void UpdateCustomFieldEditors() {
            calcAmount.Value = ((MyAppointmentFormController)base.Controller).Amount;
            tbContact.Text = ((MyAppointmentFormController)base.Controller).ContactInfo;
        }

        private void calcAmount_EditValueChanged(object sender, EventArgs e) {
            ((MyAppointmentFormController)base.Controller).Amount = calcAmount.Value;
        }

        private void tbContact_EditValueChanged(object sender, EventArgs e) {
            ((MyAppointmentFormController)base.Controller).ContactInfo = tbContact.Text;
        }
        #endregion #Form-Controller        
    }

    class MyAppointmentFormController : AppointmentFormController {
        public MyAppointmentFormController(SchedulerControl control, Appointment apt)
            : base(control, apt) {
        }
        #region #CustomFieldProperties
        public decimal Amount {
            get {
                if (base.EditedAppointmentCopy.CustomFields["Amount"] != null)
                    return Convert.ToDecimal(base.EditedAppointmentCopy.CustomFields["Amount"]);
                else return 0.0M;
            }
            set {
                base.EditedAppointmentCopy.CustomFields["Amount"] = value;
            }
        }
        public string ContactInfo {
            get {
                if (base.EditedAppointmentCopy.CustomFields["ContactInfo"] != null)
                    return base.EditedAppointmentCopy.CustomFields["ContactInfo"].ToString();
                else return String.Empty;
            }
            set {
                base.EditedAppointmentCopy.CustomFields["ContactInfo"] = value;
            }
        }

        public decimal SourceAmount {
            get {
                
                if (base.SourceAppointment.CustomFields["Amount"] != null)
                    return Convert.ToDecimal(base.SourceAppointment.CustomFields["Amount"]);
                else
                    return (decimal)0;
            }
            set {
                base.SourceAppointment.CustomFields["Amount"] = value;
            }
        }
        public string SourceContactInfo {
            get {
                if (base.SourceAppointment.CustomFields["ContactInfo"] != null)
                    return base.SourceAppointment.CustomFields["ContactInfo"].ToString();
                else
                    return String.Empty;
            }
            set {
                base.SourceAppointment.CustomFields["ContactInfo"] = value;
            }
        }
        #endregion #CustomFieldProperties

        // Indicate whether an occurrence should be converted into an exception when the Recurrence Form is invoked
        #region #IsAppointmentChanged
        public override bool IsAppointmentChanged() {
            if (base.IsAppointmentChanged())
                return true;
            return SourceAmount != Amount ||
                SourceContactInfo != ContactInfo;
        }
        #endregion #IsAppointmentChanged
        #region #ApplyCustomFieldsValues
        protected override void ApplyCustomFieldsValues() {
            base.SourceAppointment.CustomFields["Amount"] = Amount;
            base.SourceAppointment.CustomFields["ContactInfo"] = ContactInfo;
            base.ApplyCustomFieldsValues();
        }
        #endregion #ApplyCustomFieldsValues
    }
}

File: VB/SimpleCustomFields/MyAppointmentForm.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
'#Region "#usingsform"
Imports DevExpress.XtraScheduler.UI
Imports DevExpress.XtraScheduler

'#End Region  ' #usingsform
Namespace SimpleCustomFields

    Public Partial Class MyAppointmentForm
        Inherits AppointmentForm

        Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment)
            MyBase.New(control, apt)
            ' Required to initialize custom field editors.
            InitializeComponent()
        End Sub

        Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean)
            MyBase.New(control, apt, openRecurrenceForm)
            ' Required to initialize custom field editors.
            InitializeComponent()
        End Sub

        Public Overrides Sub LoadFormData(ByVal apt As Appointment)
            MyBase.LoadFormData(apt)
            UpdateCustomFieldEditors()
        End Sub

'#Region "#Form-Controller"
        Protected Overrides Function CreateController(ByVal control As SchedulerControl, ByVal apt As Appointment) As AppointmentFormController
            Return New MyAppointmentFormController(control, apt)
        End Function

        Protected Sub UpdateCustomFieldEditors()
            calcAmount.Value = CType(Controller, MyAppointmentFormController).Amount
            tbContact.Text = CType(Controller, MyAppointmentFormController).ContactInfo
        End Sub

        Private Sub calcAmount_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            CType(Controller, MyAppointmentFormController).Amount = calcAmount.Value
        End Sub

        Private Sub tbContact_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
            CType(Controller, MyAppointmentFormController).ContactInfo = tbContact.Text
        End Sub
'#End Region  ' #Form-Controller        
    End Class

    Friend Class MyAppointmentFormController
        Inherits AppointmentFormController

        Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment)
            MyBase.New(control, apt)
        End Sub

'#Region "#CustomFieldProperties"
        Public Property Amount As Decimal
            Get
                If EditedAppointmentCopy.CustomFields("Amount") IsNot Nothing Then
                    Return Convert.ToDecimal(EditedAppointmentCopy.CustomFields("Amount"))
                Else
                    Return 0.0D
                End If
            End Get

            Set(ByVal value As Decimal)
                EditedAppointmentCopy.CustomFields("Amount") = value
            End Set
        End Property

        Public Property ContactInfo As String
            Get
                If EditedAppointmentCopy.CustomFields("ContactInfo") IsNot Nothing Then
                    Return EditedAppointmentCopy.CustomFields("ContactInfo").ToString()
                Else
                    Return String.Empty
                End If
            End Get

            Set(ByVal value As String)
                EditedAppointmentCopy.CustomFields("ContactInfo") = value
            End Set
        End Property

        Public Property SourceAmount As Decimal
            Get
                If SourceAppointment.CustomFields("Amount") IsNot Nothing Then
                    Return Convert.ToDecimal(SourceAppointment.CustomFields("Amount"))
                Else
                    Return CDec(0)
                End If
            End Get

            Set(ByVal value As Decimal)
                SourceAppointment.CustomFields("Amount") = value
            End Set
        End Property

        Public Property SourceContactInfo As String
            Get
                If SourceAppointment.CustomFields("ContactInfo") IsNot Nothing Then
                    Return SourceAppointment.CustomFields("ContactInfo").ToString()
                Else
                    Return String.Empty
                End If
            End Get

            Set(ByVal value As String)
                SourceAppointment.CustomFields("ContactInfo") = value
            End Set
        End Property

'#End Region  ' #CustomFieldProperties
        ' Indicate whether an occurrence should be converted into an exception when the Recurrence Form is invoked
'#Region "#IsAppointmentChanged"
        Public Overrides Function IsAppointmentChanged() As Boolean
            If MyBase.IsAppointmentChanged() Then Return True
            Return SourceAmount <> Amount OrElse Not Equals(SourceContactInfo, ContactInfo)
        End Function

'#End Region  ' #IsAppointmentChanged
'#Region "#ApplyCustomFieldsValues"
        Protected Overrides Sub ApplyCustomFieldsValues()
            SourceAppointment.CustomFields("Amount") = Amount
            SourceAppointment.CustomFields("ContactInfo") = ContactInfo
            MyBase.ApplyCustomFieldsValues()
        End Sub
'#End Region  ' #ApplyCustomFieldsValues
    End Class
End Namespace
